mtags/src/main/scala/scala/meta/internal/metals/Docstrings.scala

```diff
@@ -54,14 +66,17 @@ class Docstrings(index: GlobalSymbolIndex) {
     }
   }
 
-  private def indexSymbolDefinition(defn: SymbolDefinition): Unit = {
+  private def indexSymbolDefinition(
```

Comment:
Can we have an explicit method to expire and use: 
```scala
class Deindexer(
    input: Input.VirtualFile
) extends ScalaMtags(input) {
  override def visitOccurrence(
      occ: SymbolOccurrence,
      sinfo: SymbolInformation,
      owner: String
  ): Unit = {
    cache.remove(occ.symbol)
  }
}

  def expireSymbolDefinition(path : AbsolutePath): Unit = {
    path.toLanguage match {
      case Language.SCALA =>
        new Deindexer(path.toInput).indexRoot()
      case _ =>
    }
  }
```
`expireSymbolDefinition` can be explicitely run from `didChange`

This is more explicit and less confusing than passing the function.

Reply:
This should be actually `ScalaDeindexer` right?
Is it okay to ignore java files in this case?